mdf_matio/__init__.py

```diff
@@ -158,11 +203,40 @@ def generate_search_index(directory: str, validate_records=True,
 
         # Loop over all produced records
         metadata = group.metadata if isinstance(group.metadata, list) else [group.metadata]
+        # Validate metadata and tweak into final MDF feedstock format
+        # Will fail if any entry fails validation - no invalid entries can be allowed
+        # TODO list:
+        #   - How should validation failures (which stop processing) be communicated?
+        #   - How should the feedstock be output? (Currently yield-ed, could be written to file)
+        #   - (Future) Should feedstock be sent to Search directly here?
+        #   - import Validator (where should it live? Toolbox?)
+        #   - schema_path: MDF schema location, also makes updates easy
+        #   - dataset_metadata: Metadata for dataset entry, will be passed in, no changes needed
+        #   - validation_params: Params for validation, will be passed in, no changes needed
+
+        # Probably need source_id in later revision
+        source_id = dataset_metadata.get("mdf", {}).get("source_id", "unknown")
+
+        vald = Validator(schema_path=schema_path)
+        # Dataset validation
+        ds_res = vald.start_dataset(dataset_metadata, validation_params)
+        if not ds_res["success"]:
```

Comment:
These `if not result["success"]` seem like exactly what Python's `try/except` blocks and exception system, in general, are designed for. Why not use those?

Reply:
Exceptions (`ValidationError`) now used.